gapic-generator-java/src/main/java/com/google/api/generator/gapic/model/GapicSnippetConfig.java

```diff
@@ -0,0 +1,409 @@
+    LinkedHashMap<String, List> configSignatureParameters = new LinkedHashMap<>();
+    for (Statement.Declaration parameter : rawConfigSignature.getParametersList()) {
+      configSignatureParameters.put(
+          parameter.getName(),
+          // TODO: test for when description is null
+          Arrays.asList(
+              parameter.getDescription(), parameter.getType().getTypeKindCase(), parameter));
+    }
+    return configSignatureParameters;
+  }
+
+  // Generates parameters for main method
+  public static LinkedHashMap<VariableExpr, AssignmentExpr> composeMainMethodArgs(
+      LinkedHashMap<String, List> configSignatureParameters) {
+    LinkedHashMap<VariableExpr, AssignmentExpr> mainMethodVariables = new LinkedHashMap<>();
+    Iterator<Map.Entry<String, List>> iterator = configSignatureParameters.entrySet().iterator();
+    while (iterator.hasNext()) {
+      Map.Entry<String, List> param = iterator.next();
+      Type.TypeKindCase paramTypeKindCase = (Type.TypeKindCase) param.getValue().get(1);
+      Statement.Declaration parameter = (Statement.Declaration) param.getValue().get(2);
+      // TODO: handle other TypeKinds; currently only handles SCALAR_TYPE for prototype
+      // Not handled yet: MapType, MessageType, RepeatedType, EnumType, BytesType
+      if (paramTypeKindCase == Type.TypeKindCase.SCALAR_TYPE) {
+        Type.ScalarType varScalarType = parameter.getType().getScalarType();
+        VariableExpr methodVar =
+            VariableExpr.builder()
+                .setVariable(
+                    Variable.builder()
+                        // convert Type to TypeNode
+                        .setType(SampleComposerUtil.convertScalarTypeToTypeNode(varScalarType))
+                        // Key is the name of the parameter
+                        .setName(JavaStyle.toLowerCamelCase(param.getKey()))
+                        .build())
+                .setIsDecl(true)
+                .build();
+        // If parameters have default values, assign them
+        if (parameter.hasValue()) {
+          AssignmentExpr methodAssignment =
+              AssignmentExpr.builder()
+                  .setVariableExpr(methodVar)
+                  .setValueExpr(
+                      ValueExpr.withValue(
+                          StringObjectValue.withValue(parameter.getValue().getStringValue())))
+                  .build();
+          mainMethodVariables.put(methodVar, methodAssignment);
+        } else {
+          Value value = NullObjectValue.create();
+          Expr valueExpr = ValueExpr.builder().setValue(value).build();
+          AssignmentExpr methodAssignment =
+              AssignmentExpr.builder().setVariableExpr(methodVar).setValueExpr(valueExpr).build();
+          mainMethodVariables.put(methodVar, methodAssignment);
+        }
+      }
+    }
+    return mainMethodVariables;
+  }
+
+  public static GapicSnippetConfig create(Optional<SnippetConfig> snippetConfigOpt) {
```

Comment:
Avoid Optional parameters. go/java-practices/optional#parameters

You can assume this method always take a value. When calling this method, use `Optional.map` https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#map-java.util.function.Function-